src/client/Model.ts

```diff
@@ -0,0 +1,115 @@
+import View from "./View";
+
+import { moveMap, MoveDirection, Player, Entity, InitMessage } from "./defs";
+import { inRange } from "../utils/util";
+
+interface IModel {
+  mapWidth: number;
+  mapHeight: number;
+  setup(data: InitMessage): void;
+  isMoveValid(direction: MoveDirection): boolean;
+  registerJoin(player: Player): void;
+  registerLeave(playerId: string): void;
+  updateTeam(updatedPlayer: Player): void;
+  updatePlayerPosition(updatedPlayer: Player): void;
+}
+
+class Model implements IModel {
+  private view: View;
+
+  private gameField: (Entity | null)[][] = [];
+  private players: Map<string, Player> = new Map();
+  private player: Player;
+
+  public mapWidth = 0;
+  public mapHeight = 0;
+
+  public constructor() {
+    this.view = new View();
+  }
+
+  public setup(data: InitMessage) {
+    this.gameField = data.gameField.map(row =>
+      row.map(rawEntity => {
+        const entity = JSON.parse(rawEntity);
+        if (entity?._entityType == "Player")
```

Comment:
Change `==` to `===`.

Reply:
_entityType is installed by the server and cannot be changed by the client in any way, so we can guarantee that this string and strict comparison will be unnecessary. But I agree to change it.

Reply:
Strict type comparing. It'll make TypeScript happier and be more safe, for sure.